rust-symcrypt/src/hash.rs

```diff
@@ -1,5 +1,24 @@
 //! Hashing functions. For further documentation please refer to symcrypt.h
 //!
+//!
+//! # Supported Hashing functions
```

Comment:
General comment about this file - can't remember if I've mentioned this before.

SymCrypt has generic hash functions, `SymCryptHash`, `SymCryptHashInit`, `SymCryptHashAppend`, and `SymCryptHashResult` which you can use with any supported hashing function. You just pass in the hash function to use as an argument. You could probably substantially reduce the amount of duplicated boilerplate code in this file by leveraging those functions. I'm not sure exactly how best to do that in Rust, but it would probably be something like having an additional trait parameter for the hash type and then mapping that to a `PCSYMCRYPT_HASH` function which you could pass into SymCrypt's generic hash functions.

Reply:
Resolved offline